File: Sum of Square Numbers/Sum_of_Square_Numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        a, b = 0, int(c**0.5) + 1
        while a <= b:
            sum_ = a * a + b * b
            if sum_ == c:
                return True
            elif sum_ > c:
                b -= 1
            else:
                a += 1
        return False
        

File: Sum of Square Numbers/Sum_of_Square_Numbers.java
class Solution {
    public boolean judgeSquareSum(int c) {
        long a = 0,b = (long)Math.sqrt(c)+1;
        while(a<=b) {
            long sum = (a*a)+(b*b);
            if(sum == c) return true;
            else if(sum > c) {
                b--;
            }
            else {
                a++;
            }
        }
        return false;
    }
}